src/components/stories/story-view/components/story-progress.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {FC, memo} from 'react';
+
+import {View} from 'react-native';
+
+import {createTheme} from '@app/helpers';
+
+import {StoryProgressItem} from './story-progress-item';
+
+import {WIDTH} from '../core/constants';
+import {StoryProgressProps} from '../core/dto/componentsDTO';
+
+const StoryProgress: FC<StoryProgressProps> = memo(
+  ({
+    progress,
+    active,
+    activeStory,
+    length,
+    progressActiveColor,
+    progressColor,
+  }) => {
+    const width =
```

Comment:
useMemo

Reply:
Length is dynamic value so memoization will not work here. We will cache all hits in this case